src/utils/hooks/use-confirm-leave.ts

```diff
@@ -0,0 +1,29 @@
+import { Router } from 'next/router'
+import { TFunction } from 'next-i18next'
+import { useEffect } from 'react'
+
+export const useConfirmOnLeavePage = (t: TFunction) => {
```

Comment:
```suggestion
export const useConfirmOnLeavePage = (message: string) => {
```

I see no point in passing entire `t` function to this hook when the only thing it actually needs is a message it should display to the user when they want to leave the page.